ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh

```diff
@@ -26,12 +23,28 @@ function cli_Login() {
     elif [[ "${CLUSTER_TYPE}" == "azure4" ]]; then
         run_command "az cloud set --name AzureCloud" || return 1
     elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
-        echo "Unsupported cluster: ${CLUSTER_TYPE}"
-        exit 1
+        AZURE_AUTH_LOCATION="${SHARED_DIR}/osServicePrincipal.json"
+        AZURESTACK_ENDPOINT=$(cat ${SHARED_DIR}/AZURESTACK_ENDPOINT)
+        SUFFIX_ENDPOINT=$(cat ${SHARED_DIR}/SUFFIX_ENDPOINT)
+        cloud_name=$(< "${CLUSTER_PROFILE_DIR}/cloud_name")
+        if [[ -f "${CLUSTER_PROFILE_DIR}/ca.pem" ]]; then
+            cp "${CLUSTER_PROFILE_DIR}/ca.pem" /tmp/ca.pem
+            cat /usr/lib64/az/lib/python*/site-packages/certifi/cacert.pem >> /tmp/ca.pem
+            export REQUESTS_CA_BUNDLE=/tmp/ca.pem
+        fi
+        az cloud register \
+            -n ${cloud_name} \
+            --endpoint-resource-manager "${AZURESTACK_ENDPOINT}" \
+            --suffix-storage-endpoint "${SUFFIX_ENDPOINT}" 
+        az cloud set -n ${cloud_name}
+        az cloud update --profile 2019-03-01-hybrid
     else
         echo "Unexpected cluster: ${CLUSTER_TYPE}"
         exit 1
     fi
+    AZURE_AUTH_CLIENT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientId)"
```

Comment:
Based on [comments](https://github.com/openshift/release/pull/36488#discussion_r1118306689), we don't need to consider dev PPE env, so no need to move those 3 lines here.

Reply:
I guess we can move them back as  AZURE_AUTH_* parameters are all set in the beginnning.